script.js

```diff
@@ -1,4 +1,235 @@
+  return `ROUND ${roundCounter}!<br><br>
+  Welcome, Player ${currentPlayer}. <br> <br> You rolled: <br> Dice 1: ${currentPlayerRolls[0]} | Dice 2: ${currentPlayerRolls[1]} <br><br> Now, please input either "1" or "2" to choose corresponding dice to be used as the first digit of your final value.</br>`;
+};
+
+var getPlayerScore = function (playerInput) {
+  var playerScore;
+  // input validation
+  if (!(playerInput == 1 || playerInput == 2)) {
+    return `Invalid input. Game will not continue until you enter either "1" or "2".<br><br>
+    You rolled: <br> Dice 1: ${currentPlayerRolls[0]} | Dice 2: ${currentPlayerRolls[1]}`;
+  }
+  // input == 1
+  else if (playerInput == 1) {
+    console.log(`Control flow: input == 1`);
+    playerScore = Number(
+      String(currentPlayerRolls[0]) + String(currentPlayerRolls[1])
+    );
+  }
+  // input == 2
+  else if (playerInput == 2) {
+    console.log(`Control flow: input == 2`);
+    playerScore = Number(
+      String(currentPlayerRolls[1]) + String(currentPlayerRolls[0])
+    );
+  }
+
+  // Store playerScore in array
+  allPlayersScore.push(playerScore);
+
+  // store overall player score until browser is refreshed
+  overallPlayersScore.push(playerScore);
+
+  // clear current player rolls array
+  currentPlayerRolls = [];
+  return `Player ${currentPlayer}, your chosen value is: ${playerScore}`;
+};
+
+var comparePlayersScores = function () {
+  var compareMessage = `Player 1 score: ${allPlayersScore[0]} <br> Player 2 score: ${allPlayersScore[1]}`;
+  // player 1 wins
+  if (allPlayersScore[0] > allPlayersScore[1]) {
+    compareMessage = `${compareMessage} <br><br> Player 1 wins!`;
+  }
+  // player 2 wins
+  if (allPlayersScore[0] < allPlayersScore[1]) {
+    compareMessage = `${compareMessage} <br><br> Player 2 wins!`;
+  }
+  // tie
+  if (allPlayersScore[0] == allPlayersScore[1]) {
+    compareMessage = `${compareMessage} <br><br> It's a tie!`;
+  }
+  return compareMessage;
+};
+
+// need a function that will calculate the sum of numbers based on when/where they were pushed in the array
+var runningSumOfNumbers = function () {
+  console.log(`adding player scores`);
+  while (i < overallPlayersScore.length) {
+    runningPlayerScore1 += overallPlayersScore[i];
+    console.log(`running player 1 score is: ${runningPlayerScore1}`);
+    i += 2;
+  }
+  while (j < overallPlayersScore.length) {
+    runningPlayerScore2 += overallPlayersScore[j];
+    console.log(`running player 2 score is: ${runningPlayerScore2}`);
+    j += 2;
+  }
+};
+
+// need a function that will store overall scores of each player
+var leaderboard = function () {
+  var rankingMessage = ``;
+  console.log(`store each player roll`);
+  while (k < overallPlayersScore.length) {
+    player1Scores.push(overallPlayersScore[k]);
+    console.log(`summary player 1 scores: ${player1Scores}`);
+    var sortedPlayer1Scores = player1Scores.slice().sort(function (a, b) {
```

Comment:
love that you have a `array.slice().sort()` here!

...there's a new syntax to this tho, `array.toSorted(function (a,b) {return b-a})`